File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

function something() {
// appRoomに代入
const appRoom = consumer.subscriptions.create("RoomChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    // return alert(data["message"]) // データ受信時にアラートとしてメッセージを表示
    const messages = document.getElementById("messages");
    messages.insertAdjacentHTML("beforeend", data["message"]);
  },

  // room_channel.rbのspeakアクションへ{ message: message }を送る
  speak: function(message) {
    return this.perform('speak', { message: message });
  }
});

window.addEventListener("keypress", (e) => {

  if (e.keyCode === 13) {
    console.log("OK");
    // KeyCode13: Enterキー
    // 入力内容をspeakメソッドへ送信
    appRoom.speak(e.target.value);
    e.target.value = "";
    e.preventDefault();
  }
});
};
window.addEventListener("load", something)

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  # サーバー接続時に実行
  def subscribed
    stream_from "room_channel"
  end

  # サーバー解除時に実行
  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end


  def speak(data)
    # パラメータdata を受信 -> room_channel.jsのreceivedメソッドへ { message: data["message"] }を送信
    # ActionCable.server.broadcast "room_channel", message: data["message"]
    Message.create! content: data["message"]
  end
end
